source/extensions/common/wasm/wasm_vm.cc

```diff
@@ -19,7 +19,17 @@ namespace Extensions {
 namespace Common {
 namespace Wasm {
 
-void EnvoyWasmVmIntegration::error(absl::string_view message) { ENVOY_LOG(trace, message); }
+proxy_wasm::LogLevel EnvoyWasmVmIntegration::getLogLevel() {
+  auto level = static_cast<size_t>(ENVOY_LOGGER().level());
+  if (level > static_cast<size_t>(proxy_wasm::LogLevel::critical)) {
```

Comment:
You should use named enums and not their underlying integer values, since they are not guaranteed to match.

This should look like:
```
switch (ENVOY_LOGGER().level()) {
case spdlog::level::trace:
  return proxy_wasm::LogLevel::trace;
...
}
```

Reply:
nice catch. thank you!